integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java

```diff
@@ -663,36 +714,172 @@ public void testStandaloneManagedRestartAlways() {
    * with spec/managedServers/0/serverStartPolicy set to IF_NEEDED.
-   * Make sure that the specified managed server is in RUNNING state.
+   * Make sure that the specified managed server is in RUNNING state
+   * Verify that if server start policy is IF_NEEDED and the server is selected
+   * to start based on the replica count, it means that server is already started or is
+   * in the process of starting. In this case, script exits without making any changes.
    */
-
   // The usecase fails NEVER->IF_NEEDED
   // https://bug.oraclecorp.com/pls/bug/webbug_print.show?c_rptno=31833260
   @Test
   @DisplayName("Restart the standalone managed server with serverStartPolicy")
   public void testStandaloneManagedRestartIfNeeded() {
-
-    String configServerPodName = domainUid + "-standalone-managed";
+    String serverName = "standalone-managed";
+    String serverPodName = domainUid + "-" + serverName;
 
     // Make sure that configured managed server is ready 
-    checkPodReadyAndServiceExists(configServerPodName,
+    checkPodReadyAndServiceExists(serverPodName,
         domainUid, domainNamespace);
     logger.info("Standalone managed server is RUNNING");
 
-    patchServerStartPolicy(
-        "/spec/managedServers/0/serverStartPolicy", "NEVER");
+    patchServerStartPolicy("/spec/managedServers/0/serverStartPolicy", "NEVER");
     logger.info("Domain is patched to shutdown standalone managed server");
 
-    checkPodDeleted(configServerPodName, domainUid, domainNamespace);
+    checkPodDeleted(serverPodName, domainUid, domainNamespace);
     logger.info("Standalone managed server shutdown success");
 
-    patchServerStartPolicy(
-        "/spec/managedServers/0/serverStartPolicy", "IF_NEEDED");
+    patchServerStartPolicy("/spec/managedServers/0/serverStartPolicy", "IF_NEEDED");
     logger.info("Domain is patched to start standalone managed server");
 
-    checkPodReadyAndServiceExists(configServerPodName,
+    checkPodReadyAndServiceExists(serverPodName,
         domainUid, domainNamespace);
     logger.info("Standalone managed server restart success");
+
+    // Verify that if server start policy is IF_NEEDED and the server is selected
+    // to start based on the replica count, it means that server is already started or is
+    // in the process of starting. In this case, script exits without making any changes.
+    String result = executeLifecycleScript(START_SERVER_SCRIPT, SERVER_LIFECYCLE, serverName);
+    assertTrue(result.contains("No changes needed"), "startServer.sh shouldn't make changes");
+  }
+
+  /**
+   * Negative test to verify:
+   * (a) the sample script can not stop or start a non-existing server
+   * (b) the sample script can not stop or start a non-existing cluster
+   * (c) the sample script can not stop or start a non-existing domain.
+   */
+  @Test
+  @DisplayName("Verify that the sample script can not stop or start non-existing components")
+  public void testRestartWrongComponent() {
```

Comment:
Change the name to testRestartNonExistingComponent()

Reply:
changed